OptimizelySDK/Odp/GraphQLManager.cs

```diff
@@ -94,15 +135,101 @@ public GraphQLManager(IErrorHandler errorHandler = null, ILogger logger = null,
         }
 
         /// <summary>
-        /// Parses JSON response
+        /// Converts the current QuerySegmentsParameters into a GraphQL query string
         /// </summary>
-        /// <param name="jsonResponse">JSON response from ODP</param>
-        /// <returns>Strongly-typed ODP Response object</returns>
-        public static Response ParseSegmentsResponseJson(string jsonResponse)
+        /// <returns>GraphQL payload</returns>
+        private static string ToGraphQLJson(string userKey, string userValue,
+            IEnumerable segmentsToCheck
+        )
+        {
+            var userValueWithEscapedQuotes = $"\\\"{userValue}\\\"";
+            var segmentsArrayJson =
+                JsonConvert.SerializeObject(segmentsToCheck).Replace("\"", "\\\"");
+
+            var json = new StringBuilder();
+            json.Append("{\"query\" : \"query {customer");
+            json.Append($"({userKey} : {userValueWithEscapedQuotes}) ");
+            json.Append("{audiences");
+            json.Append($"(subset: {segmentsArrayJson})");
+            json.Append("{edges {node {name state}}}}}\"}");
+
+            return json.ToString();
+        }
+
+        /// <summary>
+        /// Synchronous handler for querying the ODP GraphQL endpoint 
+        /// </summary>
+        /// <returns>JSON response from ODP</returns>
+        private string QuerySegments(string apiKey, string endpoint,
+            string query
+        )
         {
-            return CanBeJsonParsed(jsonResponse) ?
-                default :
-                JsonConvert.DeserializeObject<Response>(jsonResponse);
+            HttpResponseMessage response;
+            try
+            {
+                response = QuerySegmentsAsync(apiKey, endpoint, query).GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                _errorHandler.HandleError(ex);
+                _logger.Log(LogLevel.ERROR, $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (network error)");
+                return default;
+            }
+
+            var responseStatusCode = (int)response.StatusCode;
```

Comment:
Isn't this property blocking async request?

Reply:
I think I need some help checking for blocking now that I've changed so much from this review.